src/engraving/dom/edit.cpp

```diff
@@ -1587,7 +1587,7 @@ NoteVal Score::noteVal(int pitch) const
 
     // if transposing, interpret MIDI pitch as representing desired written pitch
     // set pitch based on corresponding sounding pitch
-    if (!style().styleB(Sid::concertPitch)) {
+    if (!style().styleB(Sid::concertPitch) && configuration()->pianoKeyboardUseNotatedPitch().val) {
```

Comment:
I'm not sure if we should do this check _here_. It might be preferable to move the setting back to `notationConfiguration` and retrieve its value outside the engraving module (since I'd rather consider it UI logic than business logic) and pass it as a bool parameter to `Score::addMidiPitch` and `Score::noteVal`.

Reply:
I don't disagree with what you say at all, but I have trouble seeing any philosophical difference between the Display Concert Pitch setting and this new one as far as to whether they are business logic or UI. If we are going to move the new one, we should move the old one as well, and that may get into a rats' nest of refactoring. (You would know better than I whether it is a lot.)

Or maybe the `noteVal` function should not be part of Score. But that seems like gets into *even more* refactoring.

Reply:
I would say that the difference is the scope of these settings. `Sid::concertPitch` is a score setting, and affects also how the score is typeset, i.e. really the content; `pianoKeyboardUseNotatedPitch` is a UI setting, independent of the score, affecting only the behaviour of the UI. 

There are absolutely a lot more opportunities for refactoring in the Score class and things around note input... but let's save that for different PRs!

Reply:
Hmm. I hope you will get to know me well enough that I sometimes when write out a counter argument, I then begin arguing against my own argument in my head and eventually convince myself that I was wrong. I *think* I get how Display Concert Pitch could be construed as business logic (perhaps because it controls engraving output) while there is no possibility of the input being so construed. So I will investigate your suggestion.